File: remove-duplicates-from-sorted-array/remove_duplicates_from_sorted_array.c
int removeDuplicates(int* nums, int numsSize) {
	int fore = 0, post = 1;
	for(; post < numsSize; post++){
		if( nums[fore] != nums[post] ){
			nums[++fore] = nums[post];
		}
	}

  // When nums is [], the length is not fore + 1
	return numsSize == 0 ? 0 : fore + 1;
}

File: remove-duplicates-from-sorted-array/remove_duplicates_from_sorted_array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    var fore = 0,  post = 1;
    for(; post<nums.length; post++){
        if(nums[fore] != nums[post]){
            nums[++fore] = nums[post];
        }
    }
    
    return Math.min(fore+1, nums.length);
    
};